packages/xstate-react/test/useInterpret.test.tsx

```diff
@@ -111,4 +111,52 @@ describe('useInterpret', () => {
 
     expect(actual).toEqual([1, 42]);
   });
+
+  it('should warn when machine reference is updated during the hook lifecycle', async () => {
+    console.warn = jest.fn();
+    const machine = createMachine({
+      initial: 'foo',
+      context: { id: 1 },
+      states: {
+        foo: {
+          on: {
+            CHECK: {
+              target: 'bar',
+              cond: 'hasOverflown'
+            }
+          }
+        },
+        bar: {}
+      }
+    });
+    const App = () => {
+      const [id, setId] = React.useState(1);
+      const [, send] = useMachine(
+        machine.withConfig({
+          guards: {
+            hasOverflown: () => id > 1
+          }
+        })
+      );
+
+      return (
+        <>
+          <button
+            onClick={() => {
+              setId(2);
+              send('CHECK');
+            }}
+          >
+            update id
+          </button>
+        </>
+      );
+    };
+
+    const { findByRole } = render(<App />);
+
+    (await findByRole('button')).click();
+
+    expect(console.warn).toHaveBeenCalledTimes(1);
```

Comment:
It would be great to confirm that the reported message is actually ours:
```suggestion
    expect(console.warn).toHaveBeenCalledTimes(1);
    expect(console.warn.mock.calls[0]).toMatchInlineSnapshot();
```